gips/0065-subgraph-availability-manager.md

```diff
@@ -0,0 +1,123 @@
+
+# Rationale and Alternatives
+
+The presented proposal serves as a temporary solution; we will continue our efforts to explore additional methods for further decentralization or even the complete elimination of the subgraph availability oracle. We believe this implementation provides a short-term fix that significantly improves decentralization compared to the previous 1-of-1 approach.
+
+---
+
+*This section contains the body of the Subgraph Availability Oracle Operator Charter.*
+
+# Subgraph Availability Oracle Operator Charter
+
+## 1. Role of the Subgraph Availability Oracle.
+
+The primary role of the Subgraph Availability Oracle in The Graph Network is to ensure the availability and validity of subgraph manifests. Oracles are responsible for verifying subgraphs availability and conducting validity checks.
+
+Oracles operate as part of the `SubgraphAvailabilityManager` smart contract, contributing to the decentralized decision-making process by casting votes regarding the availability and validity of subgraph files.
+
+## 2. Appointment and Governance of Subgraph Availability Oracles Operators
+
+Oracle operators are appointed by The Graph Council, which retains the authority to add or remove oracle operators, ensuring compliance with the guidelines and responsibilities outlined in this charter.
+
+The number of oracle operators is strictly limited to five, ensuring a balance between decentralization and manageability.
+
+## 3. Responsibility and Integrity
+
+Operators must ensure their oracles remain operational, acknowledging that while occasional outages are part of normal operations, efforts to minimize these are crucial. Operators should communicate their infrastructure choices with one another to prevent risks associated with common dependencies, thereby enhancing the network's overall resilience. Active participation in a group chat to discuss operational challenges and coordinate on mitigating risks is required. Operators who are unresponsive or consistently underperform may be recommended for replacement by the technical advisory board or fellow operators to the Graph Council.
```

Comment:
It might be an overhead for now but we may want to think of any contingency process that could be triggered in case any of the operators starts behaving unexpectedly.

Reply:
It's stated in the charter that operators not complying with the Charter will be subject to removal. Operators themselves are responsible for monitoring and investigating oracles behaving unexpectedly. In case an oracle is detected to be intentionally or consistently not following the Charter, operators will suggest to the Council that it should be replaced. The robustness of this system comes from the fact that it has 5 oracles and requires at least 3 to be running honestly, so even if 2 oracles are behaving dishonestly or unexpectedly, it's still operational.